compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -2688,6 +2717,21 @@ private ParserRuleContext getNextRuleForKeywords(ParserRuleContext currentCtx, i
                     endContext(); // end let-clause-let-var-decl
                 }
                 return ParserRuleContext.EXPRESSION;
+            case ORDER_KEYWORD:
+                parentCtx = getParentContext();
+                if (parentCtx == ParserRuleContext.LET_CLAUSE_LET_VAR_DECL) {
+                    endContext(); // end let-clause-let-var-decl
+                }
+                return ParserRuleContext.BY_KEYWORD;
+            case BY_KEYWORD:
+                return ParserRuleContext.ORDER_KEY;
+            case ASCENDING_KEYWORD:
+            case DESCENDING_KEYWORD:
+                parentCtx = getParentContext();
+                if (parentCtx == ParserRuleContext.ORDER_CLAUSE_END) {
```

Comment:
Looks like a new context is not started for `ORDER_CLAUSE_END`, so this condition may never be true?

I think a better approach would be to say, next token context after the ascending/descending keywords is `ORDER_KEY_RHS`. And it has alternative paths - ORDER_KEY and ORDER_CLAUSE_END